### PR TITLE
Link porting guide table to header documentation

### PR DESCRIPTION
**Pull request by samuel40791765**
### Description of changes: 
This finalizes the changes for the previous two documentation PRS: https://github.com/aws/aws-lc/pull/1463 and https://github.com/aws/aws-lc/pull/1473. Now that these two PRs are in, we have a commit to link the header documentation from the table.

### Call-outs:
N/A

### Testing:
N/A

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

File: docs/porting/configuration-differences.md [hunk 1]
```diff
@@ -259,8 +315,11 @@ The following table contains the differences in libssl configuration options AWS
   <td rowspan=2>
   <p>
     <span>
-      <a href="https://www.openssl.org/docs/manmaster/man3/X509_VERIFY_PARAM_get0_peername.html">
+      <a href="https://github.com/aws/aws-lc/blob/e91524c10ad698fd56f77289ba3430baf3c7af64/include/openssl/ssl.h#L3089-L3096l">
         SSL_set_hostflags<br>
+      </a>
+      <br>
+      <a href="https://www.openssl.org/docs/manmaster/man3/X509_VERIFY_PARAM_get0_peername.html">
```
Comment (andrewhop) [inline docs/porting/configuration-differences.md:322]: Can we add the required documentation to our header files and link to that instead of OpenSSL?

File: docs/porting/functionality-differences.md [hunk 2]
```diff
@@ -919,7 +918,7 @@ Older and less common usages of `EVP_PKEY` have been removed. For example, signi
   <td rowspan=2>
   <p>
     <span>
-        <a href="https://github.com/aws/aws-lc/blob/5c358103c5df836b9343bf995717b5bc13d5e82f/include/openssl/digest.h#L303-L310">
+        <a href="https://github.com/aws/aws-lc/blob/746d06505b3a3827cf61959ca0c3d87c3f21accc/include/openssl/digest.h#L347-L370">
             digest.h
```
Comment (andrewhop) [inline docs/porting/functionality-differences.md:922]: This link highlight includes the SHA3 flag which looks unrelated, or should this table include EVP_MD_unstable_sha3_* functions which do nothing?

File: docs/porting/functionality-differences.md [hunk 3]
```diff
@@ -1000,7 +999,7 @@ Older and less common usages of `EVP_PKEY` have been removed. For example, signi
   <td>
   <p>
     <span>
-        <a href="https://github.com/aws/aws-lc/blob/5c358103c5df836b9343bf995717b5bc13d5e82f/include/openssl/bio.h#L865-L866">
+        <a href="https://github.com/aws/aws-lc/blob/746d06505b3a3827cf61959ca0c3d87c3f21accc/include/openssl/bio.h#L865-L866">
```
Comment (andrewhop) [inline docs/porting/functionality-differences.md:1002]: This links to  BIO_set_shutdown not BIO_set_write_buffer_size

File: docs/porting/configuration-differences.md (hunk 1 above)
Comment (samuel40791765) [reply docs/porting/configuration-differences.md:322]: I opted for this only because the functions were all over the place in the headers, but did similar things. I can add documentation to these functions, but I'll have to create a follow up PR to update the links.

Comment (samuel40791765) [reply docs/porting/configuration-differences.md:322]: I noticed upstream had some commits to finish documenting `x509.h`: https://github.com/google/boringssl/commit/5bed5b9aaab4edc8c0ee62493b6e760f9f7a3457
The documentation is much more robust here, should we prioritize taking these commits instead?

**Review by torben-hansen — APPROVED** [review]

File: docs/porting/configuration-differences.md (hunk 1 above)
Comment (samuel40791765) [reply docs/porting/configuration-differences.md]: Stole some documentation from https://github.com/google/boringssl/commit/5bed5b9aaab4edc8c0ee62493b6e760f9f7a3457 prematurely and added a "TODO" to update the links once we officially take it. There's a large assortment of commits in the header file so we'll want to take these incrementally.

Updating the related functions now is better than the current state of no documentation. Will create a follow up PR to link the table to the new text.

**Review by andrewhop — APPROVED** [review]

**Review by torben-hansen — APPROVED** [review]
